### PR TITLE
proposal to add subcommands init, join, reset for yurtctl.

### PR DESCRIPTION
**Pull request by zyjhtangtang**
/kind design

#### What this PR does / why we need it:

This proposal add three subcommands `init`, `join` and `reset` for yurtctl. The subcommand `init` can create an all-in-one kubernetes cluster, simultaneously convert the kuberntes cluster to an OpenYurt cluster. The subcommand `join` is used to add a new node to an OpenYurt cluster, including cloud nodes and edge nodes. The subcommand `reset` can restore the node to the state before joining OpenYurt cluster.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

/assign @rambohe-ch @huangyuqi

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: docs/proposals/20210607-adding-subcommands-server-join-reset-for-yurtctl.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+      - [Parameters for yurtctl join](#parameters-for-yurtctl-join)
+    - [Resetting node](#resetting-node)
+  - [Implementation History](#implementation-history)
+
+## Glossary
+
+Refer to the [OpenYurt Glossary](https://github.com/openyurtio/openyurt/blob/master/docs/proposals/00_openyurt-glossary.md).
+
+## Summary
+This proposal add three subcommands `server`, `join` and `reset` for yurtctl. The subcommand `server` can create an all-in-one kubernetes cluster, simultaneously convert the kuberntes cluster to an OpenYurt cluster. The subcommand `join` is used to add a new node to an OpenYurt cluster, including cloud nodes and edge nodes. The subcommand `reset` can restore the node to the state before joining OpenYurt cluster.
```
Comment (rambohe-ch) [inline docs/proposals/20210607-adding-subcommands-server-join-reset-for-yurtctl.md:38]: how about rename the subcommand `server` to `init`, so the  experience of yurtctl will be consistent with Kubeadm

**Comment by yanyhui** [comment]
I think the 'join' subcommand should support multiple scenarios:

First, join node after executing 'server' subcommand;

Second, after the yurtctl convert to openyurt is executed on the original k8s cluster, the 'join' subcommand can be executed to access the new node;

Third, on the basis of the original k8s cluster, after manually installing openyurt, you can execute the 'join' subcommand to access the new node.

File: docs/proposals/20210607-adding-subcommands-server-join-reset-for-yurtctl.md (hunk 1 above)
Comment (zyjhtangtang) [reply docs/proposals/20210607-adding-subcommands-server-join-reset-for-yurtctl.md]: OK, I have renamed the subcommand server to init.

**Pull request merged**